File: WordCountS/src/spark/WordCount.scala
package spark

import org.apache.spark._
import SparkContext._
object WordCount {
//  def main(args:Array[String]):Unit= {
//  
//  val conf =new SparkConf().setMaster("local[*]").setAppName("word count")
//  val sc = new SparkContext(conf)
//  val textFile =sc.textFile("/Users/didi/java/WordCount/input/input1.txt")
////val result = textFile.flatMap(line => line.split("\\s+")).map(word => (word, 1)).reduceByKey(_ + _)
//  textFile.flatMap(_.split(" ")).map((_,1)).reduceByKey(_+_).collect().foreach(println)
//  }
   def main(args: Array[String]) {
//    var fileName="/Users/didi/java/WordCount/input/input1.txt" 
//     //val conf = new SparkConf()
//    val conf =new SparkConf().setMaster("local").setAppName("word count")
//    //val conf =new SparkConf().setMaster("127.0.0.1:55477").setAppName("word count")
//     val sc = new SparkContext(conf)
//     val line = sc.textFile(fileName)
//
//     line.flatMap(_.split(" ")).map((_, 1)).reduceByKey(_+_).collect().foreach(println)
//
//     sc.stop()
     //val conf =new SparkConf().setMaster("local[4]").setAppName("word count")
     //val sc = new SparkContext("local", "WordCount",
    //System.getenv("SPARK_HOME"))
     //val conf = new SparkConf().setAppName("Spakr WordCount written by jaba").setMaster("127.0.0.1:8085");
     val conf = new SparkConf()
             .setMaster("local")
             .setAppName("CountingSheep")
             .set("spark.executor.memory", "1g")
     val sc = new SparkContext(conf)
     //val file=sc.textFile("hdfs://master.Hadoop/input/*.txt")
     //val file=sc.textFile("hdfs://matser:9000/input/*.txt")
      val file=sc.textFile("hdfs://localhost:8020/input/*.txt")
     //val file=sc.textFile("file:///Users/didi/java/WordCount/input/*.txt")
     val count=file.flatMap(line=>line.split(" ")).map(word=>(word,1)).reduceByKey(_+_)
    count.collect.foreach(println)
     sc.stop()
 }
}

File: WordCoundSJ/src/WordCount.java

import java.util.Arrays;
import java.util.Iterator;
import java.util.List;
import java.util.regex.Pattern;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.FlatMapFunction;
import org.apache.spark.api.java.function.Function2;
import org.apache.spark.api.java.function.PairFunction;
import scala.Tuple2;

public class WordCount {
	@SuppressWarnings({"rawtypes","unchecked" })
	public static void main(String[] args) {
		final Pattern SPLIT = Pattern.compile(" ");

	SparkConf conf =new SparkConf().setMaster("local[1]").setAppName("word count");

	JavaSparkContext context = new JavaSparkContext(conf);

	JavaRDD<String> lines = context.textFile("/Users/wxzyhx/Downloads/paper1.txt");

	@SuppressWarnings("unchecked")
	JavaRDD words =lines.flatMap(new FlatMapFunction() {
		@Override
		public Iterator call(Object line)throws Exception {
          return (Iterator) Arrays.asList(SPLIT.split((String)line));
   }
});

	JavaPairRDD ones =words.mapToPair(new PairFunction() {
  @Override

	public Tuple2 call(Object word)throws Exception {

	return new Tuple2((String)word, 1);

	}
});

	JavaPairRDD counts =ones.reduceByKey(new Function2() {
@Override
public Integer call(Object arg0, Object arg1)throws Exception {

	return (Integer)arg0 + (Integer)arg1;

	}

	});

	List<Tuple2> output =counts.collect();

	for (Tuple2 tuple :output) {

	System.out.println(tuple._1() +": " +tuple._2());

	}

	context.close();

	}
}
